src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs

```diff
@@ -0,0 +1,232 @@
+            Contracts.Assert(returnType.IsTable);
+
+            // Source table type
+            DType sourceType = argTypes[0];
+
+            if (args[0] is AsNode)
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceScopeNotSupported);
+            }
+
+            if (sourceType.Contains(new DName(thisGroup)))
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceContainsThisGroupColumn);
+            }
+
+            var atLeastOneGroupByColumn = false;
+
+            for (int i = 1; i < args.Length; i++)
+            {
+                var argType = argTypes[i];
+                var arg = args[i];
+
+                DName columnName;
+                DType existingType;
+
+                // All args starting at index 1 need to be identifiers or aggregate functions (wrapped in AsNode node).
+                switch (arg)
+                {
+                    case AsNode nameNode:
+                        existingType = argType;
+                        columnName = nameNode.Right.Name;
+                        break;
+
+                    case FirstNameNode:
+                        if (!TryGetColumnLogicalName(sourceType, context.Features.SupportColumnNamesAsIdentifiers, arg, errors, out columnName, out existingType))
+                        {
+                            isValid = false;
+                            continue;
+                        }
+
+                        atLeastOneGroupByColumn = true;
+                        break;
+
+                    default:
+                        isValid = false;
+                        errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrSummarizeInvalidArg);
+                        continue;
+                }
+
+                if (columnName == thisGroup)
+                {
+                    isValid = false;
+                    errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrSummarizeThisGroupColumnName);
+                    continue;
+                }
+                
+                if (!existingType.IsPrimitive)
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrNeedPrimitive);
+                    isValid = false;
+                    continue;
+                }
+
+                var fError = false;
+                returnType = returnType.Add(ref fError, DPath.Root, columnName, existingType);
+                if (fError)
+                {
+                    isValid = false;
+                    errors.EnsureError(DocumentErrorSeverity.Severe, arg, TexlStrings.ErrColConflict_Name, columnName);
+                    continue;
+                }
+            }
+
+            if (!atLeastOneGroupByColumn)
```

Comment:
>if (!atLeastOneGroupByColumn) [](http://example.com/codeflow?start=12&length=29)

We check that we have at least one grouping column, but do we check if we have at least one column to group? For example, would `Summarize(MyTable, CountRows(ThisGroup) As Count, Sum(ThisGroup, 1) As SumOfOnes)` work? If it does, it shouldn't.